feat: Option to prevent propagating collision events from ShapeHitbox to _hitboxParent (#2594)
flame-engine/flame: doc/flame/collision_detection.md

a size and a position etc. The other way is to use the `relative` constructor which defines the
hitbox in relation to the size of its intended parent.

In some specific cases you might want to handle collisions only between hitboxes, without

> propagating 'onCollision*' events to hitbox's parent component. For example, a vehicle could have

Reviewer: ```suggestion
propagating `onCollision* events to the hitbox's parent component. For example, a vehicle could have a
```

> body hitbox to control collisions and side hitboxes to check availability to turn left or right.

Reviewer: ```suggestion
body hitbox to control collisions and side hitboxes to check the possibility to turn left or right.
```

> So, colliding with body hitbox means a collision with component itself, whereas colliding with side

Reviewer: ```suggestion
So, colliding with a body hitbox means colliding with the component itself, whereas colliding with a side
```

> In some specific cases you might want to handle collisions only between hitboxes, without
> propagating `onCollision*` events to the hitbox's parent component. For example, a vehicle could have a
> body hitbox to control collisions and side hitboxes to check the possibility to turn left or right.
> So, colliding with a body hitbox means colliding with the component itself, whereas colliding with a side
> hitbox does not mean a real collision and should not be propagated to hitbox's parent.
> For this case you can set `triggersParentCollision` variable to `false`:

Reviewer: ```suggestion

In some specific cases you might want to handle collisions only between hitboxes, without
propagating `onCollision*` events to the hitbox's parent component. For example, a vehicle could have a
body hitbox to control collisions and side hitboxes to check the possibility to turn left or right.
So, colliding with a body hitbox means colliding with the component itself, whereas colliding with a side
hitbox does not mean a real collision and should not be propagated to hitbox's parent.
For this case you can set `triggersParentCollision` variable to `false`:
```

a size and a position etc. The other way is to use the `relative` constructor which defines the
hitbox in relation to the size of its intended parent.

> In some specific cases you might want to handle collisions only between hitboxes, without
> propagating `onCollision*` events to the hitbox's parent component. For example, a vehicle could have a
> body hitbox to control collisions and side hitboxes to check the possibility to turn left or right.
> So, colliding with a body hitbox means colliding with the component itself, whereas colliding with a side
> hitbox does not mean a real collision and should not be propagated to hitbox's parent.
> For this case you can set `triggersParentCollision` variable to `false`:

Reviewer: ```suggestion
In some specific cases you might want to handle collisions only between hitboxes, without
propagating `onCollision*` events to the hitbox's parent component. For example, a vehicle could
have a body hitbox to control collisions and side hitboxes to check the possibility to turn left
or right.
So, colliding with a body hitbox means colliding with the component itself, whereas colliding with
a side hitbox does not mean a real collision and should not be propagated to hitbox's parent.
For this case you can set `triggersParentCollision` variable to `false`:
```
